ui/src/app/views/registration-dashboard.tsx

```diff
@@ -151,25 +159,34 @@ export class RegistrationDashboard extends React.Component<RegistrationDashboard
     super(props);
     this.state = {
       trainingWarningOpen: !props.firstVisitTraining,
-      taskCompletionMap: new Map<number, boolean>(),
```

Comment:
I bumped this out of the state, since it's directly calculated from props values. The logic is now within the getTaskCompletionList function.

Reply:
👍